### ISSUE TITLE
Things to clean from /api/configuration

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#8151]
There's no way these are needed for users:

- [ ] ftp_upload_dir
- [ ] smtp_server

![image](https://user-images.githubusercontent.com/458683/59517438-e5ff6e00-8ec3-11e9-90a9-3ee4755b23fd.png)

**Comment by martenson** [comment]
I think this url is displayed in the upload dialogue.

**Comment by hexylena** [comment]
Is that the `ftp_upload_site` you're maybe thinking of?

**Comment by martenson** [comment]
@erasche you are right

`smtp_server` is used in `/login/Register.vue`, `login/Index.vue` and other client code
`ftp_upload_dir` is *probably* useless? (yeah, we can drop it from API and templates/webapps/galaxy/galaxy.masthead.mako also I think)

**Comment by hexylena** [comment]
I was surprised to see the mail server domain and I guess not something like "smtp: true" or so, but, probably more work to change and for minimal benefit.

**Comment by dannon** [comment]
I think they should both go.  Right now the `smtp_server` is just used to see if it's configured, so I changed that to a generic boolean in #8206.  This is not something we'd ever want an external user touching directly anyway, so now we have `server_mail_configured`.

I didn't see where `ftp_upload_dir` is used at all, and that shouldn't be something that we need exposed at all, so I've removed it in that same PR.

If you find more to drop, feel free to push to that branch.
